snapshotter/demux/snapshotter.go

```diff
@@ -267,6 +267,30 @@ func (s *Snapshotter) Close() error {
 	return s.cache.Close()
 }
 
+// Cleanup proxies remote snapshotter cleanup request.
+//
+// See https://github.com/containerd/containerd/blob/main/snapshots/snapshotter.go
+func (s *Snapshotter) Cleanup(ctx context.Context) error {
+	contextLogger := log.G(ctx).WithField("function", "Cleanup")
+	namespace, err := getNamespaceFromContext(ctx, contextLogger)
+	if err != nil {
+		return err
+	}
+	logger := contextLogger.WithField("namespace", namespace)
+
+	snapshotter, err := s.getSnapshotterFromCache(ctx, namespace, logger)
+	if err != nil {
+		return err
+	}
+
+	err = snapshotter.(snapshots.Cleaner).Cleanup(ctx)
+	if err != nil {
+		contextLogger.WithError(err).Error(proxiedFunctionCallErrorString)
+		return err
+	}
```

Comment:
Wouldn't it log the error twice by this function and the function's caller?

Reply:
Yes, good catch. Good chance to simplify this. Addressed in latest revision.